ext/wry_ruby/src/application/tray_id.rs

```diff
@@ -0,0 +1,50 @@
+use magnus::{function, Error, ExceptionClass, Module, Object, RString};
+use std::cell::RefCell;
+use wry::application::TrayId as TrayIdImpl;
+
+#[derive(Clone, Debug)]
+#[magnus::wrap(class = "TrayId")]
+pub struct TrayId {
+    inner: Option<RefCell<TrayIdImpl>>,
+}
+
+impl TrayId {
+    pub const EMPTY: TrayId = TrayId { inner: None };
+
+    pub fn new(unique_string: RString) -> Result<Self, Error> {
+        if unique_string.is_empty() {
+            return Ok(TrayId::EMPTY);
+        }
+
+        let s = match unique_string.as_interned_str() {
+            Some(str_ref) => str_ref.as_str().unwrap(),
+            None => {
+                println!("ok why is it none, hm check freeze or no");
+                return Err(Error::new(
+                    ExceptionClass::default(),
+                    "Interned string not available",
+                ));
+            }
+        };
+
+        Ok(Self {
+            inner: Some(RefCell::new(TrayIdImpl::new(s))),
+        })
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.inner.is_none()
+    }
+    
+}
+
+pub fn init() -> Result<(), Error> {
+    let class = magnus::define_class("TrayId", Default::default())?;
+
+    class.define_singleton_method("new", function!(TrayId::new, 1))?;
+    class.define_method("is_empty", function!(TrayId::is_empty, 1))?;
```

Comment:
here instead of `function!` we should use `method!` which allows this to be an actual instance method

Reply:
okay